lib/fetchJson.ts

```diff
@@ -2,6 +2,13 @@ export async function fetchJson(url: RequestInfo, init?: RequestInit): Promise<a
   const response = await fetch(url, init);
   const json = await response.json();
 
-  if (!response.ok) throw new Error(`${response.statusText}: ${json.error?.message || JSON.stringify(json)}`);
+  if (!response.ok) {
+    return Promise.reject({
```

Comment:
Nice, this was driving me crazy. This is a better way to handle it, just have to make sure our implementations will handle this schema.